rskj-core/src/main/java/co/rsk/core/bc/TransactionPoolImpl.java

```diff
@@ -252,14 +264,25 @@ public synchronized TransactionPoolAddResult addTransaction(final Transaction tx
 
         pendingTransactions.addTransaction(tx);
 
-        if (listener != null) {
-            EventDispatchThread.invokeLater(() -> {
-                listener.onPendingTransactionsReceived(Collections.singletonList(tx));
-                listener.onTransactionPoolChanged(TransactionPoolImpl.this);
-            });
+        return TransactionPoolAddResult.ok();
+    }
+
+    @Override
+    public synchronized TransactionPoolAddResult addTransaction(final Transaction tx) {
```

Comment:
I would rewrite this to call addTransactions with a list of one to avoid duplicated logic, but it's a minor detail

Reply:
Well, the `addTransactions` method returns a list of transaction, that it was the original intention of the method. Maybe we could use this results in the (possible) removal of `TransactionGateway`, to return to the original behavior: `TransactionPool` returns the added transactions, to be relayed to the other nodes by the caller object. So, we would visit this refactor in that upcoming issue